google/cloud/bigquery/integration_tests/bigquery_read_integration_test.cc

```diff
@@ -80,6 +81,19 @@ std::int64_t CountRowsFromStream(
   return num_rows;
 }
 
+template <typename Functor>
+google::cloud::internal::invoke_result_t<Functor> StatusRetryLoop(
+    Functor&& operation,
+    std::vector<int> const& delay_seconds = {10, 30, 60, 60}) {
+  google::cloud::internal::invoke_result_t<Functor> status;
+  for (auto delay : delay_seconds) {
+    status = operation();
+    if (status.ok()) break;
```

Comment:
Q. If we know that `status.ok()` is a valid expression, should we not abstract away `Status` from the functor signature?